From 0521d90c3d081bc45aad4815698884ae32e35041 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 10 Sep 2014 16:02:58 -0400
Subject: [PATCH 09/18] Get rid of an invalid comparison.

Covscan.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/efibootmgr/efibootmgr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/efibootmgr/efibootmgr.c b/src/efibootmgr/efibootmgr.c
index 88a4ef7..5280180 100644
--- a/src/efibootmgr/efibootmgr.c
+++ b/src/efibootmgr/efibootmgr.c
@@ -1008,7 +1008,7 @@ parse_opts(int argc, char **argv)
 			break;
 		case 'b':
 			rc = sscanf(optarg, "%X", &num);
-			if (rc == 1 && num >= 0 && num < 0xffff) {
+			if (rc == 1 && num < 0xffff) {
 				opts.bootnum = num;
 			} else {
 				fprintf (stderr,"invalid hex value %s\n",optarg);
-- 
1.9.3

