From 3294351e275e01821673be29c972600737e2c5ae Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 10 Sep 2014 16:02:58 -0400
Subject: [PATCH 10/31] Get rid of an invalid comparison.

Covscan.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/efibootmgr/efibootmgr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/efibootmgr/efibootmgr.c b/src/efibootmgr/efibootmgr.c
index 0a6ca5a..d37227c 100644
--- a/src/efibootmgr/efibootmgr.c
+++ b/src/efibootmgr/efibootmgr.c
@@ -1008,7 +1008,7 @@ parse_opts(int argc, char **argv)
 			break;
 		case 'b':
 			rc = sscanf(optarg, "%X", &num);
-			if (rc == 1 && num >= 0 && num < 0xffff) {
+			if (rc == 1 && num < 0xffff) {
 				opts.bootnum = num;
 			} else {
 				fprintf (stderr,"invalid hex value %s\n",optarg);
-- 
2.7.4

