From f8a8d5d0a20ca118397e82ed81cf3b0ce492d8f4 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 10 Sep 2014 16:50:08 -0400
Subject: [PATCH 17/31] Check open()s return correctly.

Covscan.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/lib/disk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/lib/disk.c b/src/lib/disk.c
index 4536a67..337909e 100644
--- a/src/lib/disk.c
+++ b/src/lib/disk.c
@@ -328,7 +328,7 @@ disk_get_ide_pci(int fd,
 	sprintf(procname, "/proc/ide/ide%d/config", info.controllernum);
 
 	procfd = open(procname, O_RDONLY);
-	if (!procfd) {
+	if (procfd < 0) {
 		perror("opening /proc/ide/ide*/config");
 		return 1;
 	}
-- 
2.7.4

